src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -1426,14 +1425,19 @@ internal enum ErrorCode
         ERR_RefReturningCallAndAwait = 8178,
         #endregion diagnostics for ref locals and ref returns introduced in C# 7
 
+        #region stragglers for C# 7
         ERR_PredefinedValueTupleTypeNotFound = 8179,
         ERR_SemiOrLBraceOrArrowExpected = 8180,
         ERR_NewWithTupleTypeSyntax = 8181,
         ERR_PredefinedValueTupleTypeMustBeStruct = 8182,
         ERR_DiscardTypeInferenceFailed = 8183,
-        ERR_MixedDeconstructionDisallowed = 8184,
+        ERR_MixedDeconstructionUnsupported = 8184,
+        ERR_DeclarationExpressionNotPermitted = 8185,
+        ERR_MustDeclareForeachIteration = 8186,
+        ERR_TupleElementNamesInDeconstruction = 8187,
```

Comment:
For ERR_MixedDeconstructionUnsupported: four tests in Emit\CodeGen\CodeGenDeconstructTests.cs, and two in Syntax\Parsing\DeconstructionTests.cs

for ERR_DeclarationExpressionNotPermitted: many tests in Emit\CodeGen\CodeGenTupleTest.cs, Semantic\Semantics\DeconstructionTests.cs, Semantic\Semantics\LocalFunctionTests.cs, Semantic\Semantics\OutVarTests.cs, and Semantic\Semantics\PatternSwitchTests.cs

for ERR_MustDeclareForeachIteration: two tests in Emit\CodeGen\CodeGenDeconstructTests.cs

for ERR_TupleElementNamesInDeconstruction: two tests in Emit\CodeGen\CodeGenDeconstructTests.cs

Are you saying that you don't think these tests are sufficient?

---
In reply to: [89845938](https://github.com/dotnet/roslyn/pull/15548#discussion_r89845938) [](ancestors = 89845938)